### PR TITLE
[Suggestion] Updating configs and profiles

### PR DESCRIPTION
**Pull request by Mxrcon**
# Updating configs and profiles
 
Hey @abhi18av I took some time to update the nextflow.config file, i removed the configuration from the module and added them as a standard profile. I also created configuration files to each profile, a taskfile and aiming on more code readability i added workflows as inclusions to the main.nf, as this would make easier to understand that they're different workflows with different proposes.

## Pull request content
* nextflow.config
* Azure, Aws, GCP and stub profiles
* taskfile
* workflows files

# PR ending

If you have any doubt, critic or suggestion, fell free to contact me. I'm putting myself at your disposal to made any changes that you judge necessarily. 

Hope we have a merge! 
Regards, Davi

### CODE REVIEW

File: main.nf [hunk 1]
```diff
@@ -14,88 +15,14 @@ include { QUAST as QUAST_SPADES } from "./modules/quast/quast.nf" addParams(resu
...
-
-
+// Workflow inclusion
+include { BASE } from "./workflows/base.nf"
+include { QUALITY_CHECK } from "./workflows/quality_check.nf"
+include { WF_SNIPPY } from "./workflows/wf_snippy.nf"
+include { SNIPPY_ANISA_WF } from "./workflow/snippy_anisa_wf.nf"
+include { SPADES_WF } from "./workflow/spades_wf.nf"
+include { UNICYCLER_WF } from "./workflow/unicycler_wf.nf"
+include { QUAST_WF } from "./workflow/quast_wf.nf"
```
Comment (abhi18av) [inline main.nf]: @Mxrcon , could you please refactor them to make them follow a consistent naming pattern. For this project (_only_), let's follow the convention of having `_WF` at the end of all sub-workflow names.

File: modules/fastani/fastani.nf [hunk 2]
```diff
@@ -1,40 +0,0 @@
-// TODO
-
-nextflow.enable.dsl = 2
-
-params.resultsDir = "${params.outdir}/unicycler"
```
Comment (abhi18av) [inline modules/fastani/fastani.nf]: Have you removed these modules?

I do intend to complete the `bwa` and `fastani` in future over here.

File: modules/samtools/samtools.nf [hunk 3]
```diff
@@ -1,40 +0,0 @@
-// TODO
-
-nextflow.enable.dsl = 2
```
Comment (abhi18av) [inline modules/samtools/samtools.nf]: Please don't delete this module either. Replacing with a simple `TODO` would be enough.

**Review by abhi18av — CHANGES_REQUESTED** [review]
Thanks @Mxrcon for the wondeful work here, I've requested a few changes and I think we're good to go after that.

File: main.nf (hunk 1 above)
Comment (Mxrcon) [reply main.nf]: ok! I'll standardize it now!

File: modules/fastani/fastani.nf (hunk 2 above)
Comment (Mxrcon) [reply modules/fastani/fastani.nf]: yes, sorry! I'll restore them now.

File: modules/samtools/samtools.nf (hunk 3 above)
Comment (Mxrcon) [reply modules/samtools/samtools.nf]: ok! I'll restore it too

**Review by abhi18av — APPROVED** [review]
@Mxrcon , looks good to merge 👍 

Thanks Davi!

**Pull request merged**